internal/pkg/cli/pipeline_deploy.go

```diff
@@ -62,21 +66,25 @@ type deployPipelineVars struct {
 	appName          string
 	name             string
 	skipConfirmation bool
+	showDiff         bool
 }
 
 type deployPipelineOpts struct {
 	deployPipelineVars
 
-	pipelineDeployer                pipelineDeployer
-	sel                             wsPipelineSelector
-	prog                            progress
-	prompt                          prompter
-	region                          string
-	store                           store
-	ws                              wsPipelineReader
-	codestar                        codestar
-	newSvcListCmd                   func(io.Writer, string) cmd
-	newJobListCmd                   func(io.Writer, string) cmd
+	pipelineDeployer    pipelineDeployer
+	sel                 wsPipelineSelector
+	prog                progress
+	prompt              prompter
+	region              string
+	store               store
+	ws                  wsPipelineReader
+	codestar            codestar
+	writer              io.Writer
```

Comment:
nit, to keep it consistent with `svc deploy --diff`:
```suggestion
	diffWriter              io.Writer
```